chore: update pre-commit and fix linter errors

* related: https://github.com/intel/cve-bin-tool/pull/3194

We have an automated script that updates the versions of linters and tools we use in `pre-commit` and if we upgrade flake8 to 6.1.0 it gives us a bunch of errors:

```python
cve_bin_tool/cvedb.py:642:12: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
test/test_checkers.py:[32](https://github.com/intel/cve-bin-tool/actions/runs/5753663775/job/15597350652?pr=3194#step:7:33):16: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
test/test_checkers.py:33:16: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
test/test_checkers.py:34:16: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
test/test_checkers.py:35:16: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
test/test_checkers.py:36:16: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
```

These should be pretty simple mechanical things to fix but it's been a few weeks and I haven't gotten around to it, so I'm opening this up as a beginner-friendly bug for someone to work on.

You need to:

- Re-use the changes from https://github.com/intel/cve-bin-tool/pull/3194 to update .pre-commit-config.yaml and dev-requirements.txt. The important one to change is flake8 but you might as well update them all.
- resolve any flake8 errors found when the new version is run (If you're not sure how to use pre-commit to run these, there's instructions here: https://github.com/intel/cve-bin-tool/blob/main/CONTRIBUTING.md#running-linters

**Short tips for new contributors:**

* [cve-bin-tool's contributor docs](https://github.com/intel/cve-bin-tool/blob/main/CONTRIBUTING.md)
* If you've contributed to open source but not this project, you might just want our [checklist for a great pull request](https://github.com/intel/cve-bin-tool/blob/main/CONTRIBUTING.md#checklist-for-a-great-pull-request)
* cve-bin-tool uses <https://www.conventionalcommits.org/> style for commit messages, and we have a test that checks the title of your pull request (PR).  A good potential title for this one is in the title of this issue.
* You can make an issue auto close by including a comment "fixes #ISSUENUMBER" in your PR comments where ISSUENUMBER is the actual number of the issue.  This "links" the issue to the pull request.

**Claiming issues:**

* You do not need to have an issue assigned to you before you work on it.  To "claim" an issue either make a linked pull request or comment on the issue saying you'll be working on it.  
* If someone else has already commented or opened a pull request, assume it is claimed and find another issue to work on.  
* If it's been more than 1 week without progress, you can ask in a comment if the claimant is still working on it before claiming it yourself (give them at least 3 days to respond before assuming they have moved on).

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -5,18 +5,18 @@ repos:
     - id: isort
 
 -   repo: https://github.com/python/black
-    rev: 23.3.0
+    rev: 23.7.0
     hooks:
     - id: black
 
 -   repo: https://github.com/asottile/pyupgrade
-    rev: v3.7.0
+    rev: v3.10.1
     hooks:
     - id: pyupgrade
       args: ["--py38-plus"]
 
 -   repo: https://github.com/pycqa/flake8
-    rev: 6.0.0
+    rev: 6.1.0
     hooks:
     - id: flake8
       exclude: ^fuzz/generated/

--- a/cve_bin_tool/cvedb.py
+++ b/cve_bin_tool/cvedb.py
@@ -656,7 +656,7 @@ class CVEDB:
         """
 
         # For python package checkers we don't need the progress bar running
-        if type(package_names) != list:
+        if type(package_names) is not list:
             cursor.execute(query, [package_names])
             vendors = list(map(lambda x: x[0], cursor.fetchall()))
 

--- a/test/test_checkers.py
+++ b/test/test_checkers.py
@@ -29,11 +29,11 @@ class TestCheckerClass:
             VENDOR_PRODUCT = [("myvendor", "myproduct")]
             IGNORE_PATTERNS = [r"ignore"]
 
-        assert type(MyChecker.CONTAINS_PATTERNS[0]) == Pattern
-        assert type(MyChecker.VERSION_PATTERNS[0]) == Pattern
-        assert type(MyChecker.FILENAME_PATTERNS[0]) == Pattern
-        assert type(MyChecker.VENDOR_PRODUCT[0]) == VendorProductPair
-        assert type(MyChecker.IGNORE_PATTERNS[0]) == Pattern
+        assert type(MyChecker.CONTAINS_PATTERNS[0]) is Pattern
+        assert type(MyChecker.VERSION_PATTERNS[0]) is Pattern
+        assert type(MyChecker.FILENAME_PATTERNS[0]) is Pattern
+        assert type(MyChecker.VENDOR_PRODUCT[0]) is VendorProductPair
+        assert type(MyChecker.IGNORE_PATTERNS[0]) is Pattern
 
     def test_no_vpkg(self):
         with pytest.raises(AssertionError) as e:

